src/Symfony/Component/Console/Messenger/ExecuteCommandHandler.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Messenger;
+
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Input\StringInput;
+use Symfony\Component\Console\Output\BufferedOutput;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ */
+final class ExecuteCommandHandler
+{
+    public function __construct(private readonly Application $application)
+    {
+    }
+
+    public function __invoke(ExecuteCommand $message): BufferedOutput
+    {
+        $input = new StringInput($message->input);
+        $output = new BufferedOutput();
+
+        $this->application->setCatchExceptions($message->catchExceptions);
+        $this->application->run($input, $output);
+
+        return $output;
```

Comment:
Any better ideas on what to return here? `BufferedOutput` doesn't contain the exit code.

Reply:
this would mostly be excuted async isnt it?

Reply:
we are currently simply logging error output, success output seems irrelevant?

Reply:
i assume most cases run async and ignore the return value anyway

we rely on failure/error logging, and IMHO, if such data needs to to be persisted in a SQL database, it should be done by an actual [logger](https://github.com/gupalo/monolog-dbal-logger)

for those cases that consume the return value a simple array sounds fine, but feel free to make it a value object sure :)